### PR TITLE
RDK-30110: Build in release mode without errors

### PR DESCRIPTION
**Pull request by TeknoVenus**
### Description
Allow Dobby to build in Release mode without any errors. Set the log levels appropriately. 

Note release builds do not include DobbyTool or DobbyBundleGenerator.

Also renames the Xi6 platform to `Generic` as Dobby Spec support is not required on Xi6 platforms so no extra settings are needed specific to Xi6. 

Legacy Components default to OFF

Adds a new CMake option for use by Bitbake to append the SRC_REV to the version number of Dobby for easy checking of build version on the STB

### Test Procedure
Build Dobby in release mode both with and without LTO enabled. Should build without errors and no regressions should be visible in Sky and generic RDK builds

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (doesn't fit into the above categories - e.g. documentation updates)

### Requires Bitbake Recipe changes?
- [ ] The base Bitbake recipe (`meta-rdk-ext/recipes-containers/dobby/dobby.bb`) must be modified to support the changes in this PR (beyond updating `SRC_REV`)

### CODE REVIEW

File: daemon/process/settings/dobby.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (goruklu) [inline daemon/process/settings/dobby.json:1]: @TeknoVenus  dobby.json is used only when using dobbyspec.json and not the OCI bundle, correct ?

File: daemon/process/CMakeLists.txt [hunk 2]
```diff
@@ -108,8 +108,9 @@ install( # Specifies the target to install
 
 
 # Install the config file for the target platform
-if( RDK_PLATFORM STREQUAL "XI6" )
-        set( DOBBY_SETTINGS_FILE "settings/dobby.xi6.json" )
+# Generic is just a standard build with no extra settings
```
Comment (goruklu) [inline daemon/process/CMakeLists.txt:111]: I think we should keep the xi6 specific configuration even though it's not used in current builds.

**Review by goruklu — COMMENTED** [review]
The changes look good to me but I'd like to keep Xi6 config in as reference in addition to the generic, Llama and Xione configurations.

File: daemon/process/settings/dobby.json (hunk 1 above)
Comment (TeknoVenus) [reply daemon/process/settings/dobby.json:1]: The `paths`, `logging` and `network` settings are used to configure the daemon, and apply for bundle and dobbyspec containers. Those settings should always be present

The remaining settings only apply to dobbyspec containers, so are not necessary on non-sky builds (hence why I removed them from the generic RDK config)

**Review by Koczek9 — DISMISSED** [review]
LGTM

**Comment by TeknoVenus** [comment]
@Koczek9 - Git commits went weird, but made a small change in e5b71b2 to allow enabling DobbyTool in release builds - please re-review when you get chance.

**Review by goruklu — APPROVED** [review]

**Pull request merged**